source/MaterialXRuntime/Private/PvtAttribute.cpp

```diff
@@ -48,6 +48,20 @@ bool PvtOutput::isConnectable(const PvtInput* other) const
         // instead of strictly an exact match.
         return false;
     }
+    // If this is a nodegraph socket being connected to a uniform make sure
+    // the corresponding input on the outer nodegraph is also uniform.
+    if (isSocket() && other->isUniform())
```

Comment:
Checking this is just to expose and input as a socket, and there is no such thing as checking if an output can be connected to an input as outputs won't ever be uniforms (unless you tag them so on input addition which does not seem to be the case)

Reply:
Yes the only time isSocket() will return true here is when the "this" instance is a graph input socket. Ordinary outputs are never marked as uniform.